### PR TITLE
(0.87.4) Add missing `with_advective_forcing` method

### PR DESCRIPTION
**Pull request by glwagner**
Begs the question of whether this code was tested? When users pass a tuple to `forcings`, we wrap it inside `MultipleForcings`. Thus we need this method to unwrap the tuple when adding advective forcings to the total velocity field.

@jagoosw @simone-silvestri 

Should we add a test?

### CODE REVIEW

**Comment by navidcy** [comment]
Voting "yes" for adding a test.

**Comment by glwagner** [comment]
Are there any existing tests @simone-silvestri @jagoosw ?

**Comment by jagoosw** [comment]
Multiple forcing is tested but all the forcing tests just check that they don't error, so I expect in this situation the advective forcing would have no effect but wouldn't error.

**Comment by glwagner** [comment]
> Multiple forcing is tested but all the forcing tests just check that they don't error, so I expect in this situation the advective forcing would have no effect but wouldn't error.
> 
> Perhaps all of the forcing tests should really have something checking that they are changed, the forcing a could all just be set to return 1 and then we check that all of the tracers that are forced are not zero at the end?

Mm yeah I can extend the existing test to pass only if the tracer distribution changes after that time-step is taken.

File: test/test_forcings.jl [hunk 1]
```diff
@@ -119,19 +119,28 @@ function advective_and_multiple_forcing(arch)
...
+    # Time-step without an error?
     time_step!(model, 1, euler=true)
 
-    return true
-end
+    # Does `b` change?
+    a₁ = Array(interior(model.tracers.a))
+    b₁ = Array(interior(model.tracers.b))
 
+    return a₁ ≠ a₀ && b₁ ≠ b₀
```
Comment (jagoosw) [inline test/test_forcings.jl:142]: In this test do we maybe want to somehow check that both of the forcings are being done, perhaps by a two step thing like doing this test, then changing `simple_forcing` to return 1 and setting the slip velocity to 0 and testing again?

Comment (glwagner) [reply test/test_forcings.jl:142]: We can add a third tracer

**Comment by glwagner** [comment]
Ok @jagoosw I added the test you asked for, let me know what you think.

**Comment by jagoosw** [comment]
Thanks, this looks good and I can't think of cases where the test would erroneously pass now.

**Review by jagoosw — APPROVED** [review]

**Pull request merged**